### PR TITLE
Fix crash that partition table has no encoding attributes for new col…

### PR DESCRIPTION
**Pull request by gfphoenix78**
…umns

The encoding attributes are not added for new columns added by alter table. The parent hasn't any information on the new columns. Its encoding is determined by the storage type, or default encoding.

Fixes #[761](https://github.com/apache/cloudberry/issues/761)

### Type of Change
- [x] Bug fix (non-breaking change)
- [ ] New feature (non-breaking change)
- [ ] Breaking change (fix or feature with breaking changes)
- [ ] Documentation update

### Test Plan

- [ ] Unit tests added/updated
- [x] Integration tests added/updated
- [ ] Passed `make installcheck`
- [ ] Passed `make -C src/test installcheck-cbdb-parallel`

### Impact

**Performance:**

**User-facing changes:**

**Dependencies:**

### Checklist
- [ ] Followed [contribution guide](https://cloudberry.apache.org/contribute/code)
- [ ] Added/updated documentation
- [ ] Reviewed code for security implications
- [ ] Requested review from [cloudberry committers](https://github.com/orgs/apache/teams/cloudberry-committers)

### CI Skip Instructions

---

### CODE REVIEW

**Review by avamingli — DISMISSED** [review]

File: src/backend/access/aocs/aocsam_handler.c [hunk 1]
```diff
@@ -2310,6 +2310,15 @@ aoco_transform_column_encoding_clauses(Relation rel, List *aocoColumnEncoding,
...
 	NameData	compresstype_nd;
 
+	/*
+	 * The relam of partition table may be ao table, but partition table
+	 * has no entry in pg_appendonly. It shouldn't fetch encoding options
+	 * from here for partition tables. See details in function
+	 * transformColumnEncoding.
+	 */
+	if (rel && rel->rd_rel->relkind == RELKIND_PARTITIONED_TABLE)
+		rel = NULL;
```
Comment (yjhjstz) [inline src/backend/access/aocs/aocsam_handler.c:2320]: use rel = NULL to control the logical is not the best way.

Comment (gfphoenix78) [reply src/backend/access/aocs/aocsam_handler.c:2320]: Do you have any suggestions?
Whether rel is null has different code branch in the following code. It is ao-specific implementation. AO partition tables have no entry in pg_appendonly. We should avoid touch pg_appendonly for partition table.

File: src/backend/access/common/reloptions_gp.c [hunk 2]
```diff
@@ -1689,9 +1689,10 @@ transformColumnEncoding(const TableAmRoutine *tam, Relation rel, List *colDefs,
...
 				else if (!explicitOnly)
 				{
-					if (parentenc)
+					ColumnReferenceStorageDirective *parent_col_encoding;
+					parent_col_encoding = find_crsd(d->colname, parentenc);
+					if (parent_col_encoding)
 					{
-						ColumnReferenceStorageDirective *parent_col_encoding = find_crsd(d->colname, parentenc);
 						encoding = transformStorageEncodingClause(parent_col_encoding->encoding, true);
 					}
```
Comment (yjhjstz) [inline src/backend/access/common/reloptions_gp.c:1697]: when parent_col_encoding is null, what's encoding ?

Comment (gfphoenix78) [reply src/backend/access/common/reloptions_gp.c:1697]: `parent_col_encoding ` is null when adding new column for a table.

**Review by yjhjstz — APPROVED** [review]

**Review by avamingli — APPROVED** [review]

**Pull request merged**